src/qibocal/auto/execute.py

```
@@ -18,17 +20,36 @@ class Executor:
     """The graph to be executed."""
     history: History
     """The execution history, with results and exit states."""
+    qubits: Qubits
+    """Qubits to be calibrated."""
+    platform: AbstractPlatform
+    """Qubits' platform."""
+    folder: Path
+    """Output path."""
     head: Optional[Id] = None
     """The current position."""
     pending: Set[Id] = field(default_factory=set)
     """The branched off tasks, not yet executed."""
 
+    # TODO: find a more elegant way to pass everything
```

You can make the other arguments `Optional`, and set their default value to `None`.

Of course, if a run is attempted, and they are still `None`, you should raise an error :)

---

You are right. Thank you!